microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md

```diff
@@ -166,17 +166,215 @@ The [Microsoft Edge Tools for Visual Studio Code extension][VisualstudioMarketpl
+   Filter experiments in the **Experiments** setting  
+:::image-end:::  
+
+### New Vary Header column in the Cache storage pane  
+
+Use the new `Vary Header` column in the **Cache Storage** pane to display the [Vary][HttpwgSpecsRfc7231HtmlHeaderVary] HTTP response header values.  To review the history of this feature in the Chromium open-source project, navigate to Issue [1186049][CR1186049].  
+
+:::image type="complex" source="../../media/2021/04/application-cache-cache-storage-highlighted-vary-header.msft.png" alt-text="Vary Header column" lightbox="../../media/2021/04/application-cache-cache-storage-highlighted-vary-header.msft.png":::
+   Vary Header column  
+:::image-end:::  
+
+### Sources tool improvements  
+
+#### Support for new JavaScript features  
+
+DevTools now support the new [Private brand checks a.k.a. #foo in obj][V8DevFeaturesPrivateBrandChecks] JavaScript language feature, a.k.a `#foo in obj`.  The private brand checks feature extends the [in operator][MdnDocsWebJavascriptReferenceOperatorsIn] to support the [Private class fields][V8DevFeaturesClassFieldsPrivateClassFields] to test on a specific object.  Try it in the **Console** and **Sources** tools.  Also, to inspect the private fields, complete the following actions.  
+
+1.  Navigate to **debugger** pane.  
+1.  Navigate to the **Scope** section.  
+    
+To review the history of this feature in the Chromium open-source project, navigate to Issue [11374][CR11374].  
+
+<!--  
+:::image type="complex" source="../../media/2021/04/dPDCek3EhZgLQPGtEG3y0fTn4v82/RCw19qd1bZOeNFydsUCW.msft.png" alt-text="JavaScript private brand checks" lightbox="../../media/2021/04/dPDCek3EhZgLQPGtEG3y0fTn4v82/RCw19qd1bZOeNFydsUCW.msft.png":::
+   JavaScript private brand checks  
+:::image-end:::  
+-->  
+
+#### Enhanced support for breakpoints debugging  
+
+In Microsoft Edge version 90 or earlier, DevTools only set breakpoints on one raw location.  In Microsoft Edge version 91 or later, DevTools properly set breakpoints in multiple scripts.  This feature is a modern support code splitting feature.  Multiple routes may include one shared component and are named code splits.  JavaScript bundlers include [Webpack][WebpackJsMain], and [Rollup][RollupjsMain].  In Microsoft Edge version 91 or later, DevTools sets breakpoints in all relevant locations correctly.  To review the history of this feature in the Chromium open-source project, navigate to Issues [1142705][CR1142705], [979000][CR979000], and [1180794][CR1180794].  
+
+#### Support hover preview with [] notation  
+
+DevTools now support hover preview on JavaScript member expressions that use the `[]` notation in the **Sources** tool.  To review the history of this feature in the Chromium open-source project, navigate to Issue [1178305][CR1178305].  
+
+<!--  
```

Comment:
For this image, navigate to: https://codepen.io/zoherghadyali/full/jOBOBBX. Open DevTools > **Sources**. Select `Ctrl`+`P` and open **pen.js**. Set a breakpoint on line 4 and refresh the page. Hover over `arr[i]` and you should see the hover display the correct value of `a` in the array.

Reply:
Here's what I got: 
![image](https://user-images.githubusercontent.com/34316555/117218288-bc050c80-adb7-11eb-8b7c-78289a52a0ee.png)

Reply:
Updated.